packages/splits-sdk/src/client/splitV2.ts

```diff
@@ -0,0 +1,868 @@
+import {
+  Address,
+  Chain,
+  GetContractReturnType,
+  Hex,
+  Log,
+  PublicClient,
+  Transport,
+  decodeEventLog,
+  encodeEventTopics,
+  zeroAddress,
+  getContract,
+  TypedDataDomain,
+} from 'viem'
+import { splitV2ABI } from '../constants/abi/splitV2'
+import { splitV2FactoryABI } from '../constants/abi/splitV2Factory'
+import {
+  CallData,
+  CreateSplitV2Config,
+  SetPausedConfig,
+  SplitV2ExecCallsConfig,
+  SplitV2Type,
+  SplitsClientConfig,
+  TransactionConfig,
+  TransactionFormat,
+  TransferOwnershipConfig,
+  UpdateSplitV2Config,
+} from '../types'
+import {
+  BaseClientMixin,
+  BaseGasEstimatesMixin,
+  BaseTransactions,
+} from './base'
+import { applyMixins } from './mixin'
+import {
+  PERCENTAGE_SCALE,
+  SPLITS_SUPPORTED_CHAIN_IDS,
+  TransactionType,
+  getSplitV2FactoryAddress,
+} from '../constants'
+import {
+  validateAddress,
+  getNumberFromPercent,
+  getAddressAndAllocationFromRecipients,
+} from '../utils'
+import { TransactionFailedError, UnsupportedChainIdError } from '../errors'
+
+type SplitFactoryABI = typeof splitV2FactoryABI
+type SplitV2ABI = typeof splitV2ABI
+
+const VALID_ERC1271_SIG = '0x1626ba7e'
+
+// TODO:add validation to execute contract function
+class SplitV2Transactions extends BaseTransactions {
+  constructor({
+    transactionType,
+    chainId,
+    publicClient,
+    ensPublicClient,
+    walletClient,
+    includeEnsNames = false,
+  }: SplitsClientConfig & TransactionConfig) {
+    super({
+      transactionType,
+      chainId,
+      publicClient,
+      ensPublicClient,
+      walletClient,
+      includeEnsNames,
+    })
+  }
+
+  protected async _createSplit({
+    recipients,
+    distributorFeePercent,
+    splitType = SplitV2Type.Pull,
+    controller = zeroAddress,
+    creator = zeroAddress,
+    salt,
+    transactionOverrides = {},
+  }: CreateSplitV2Config): Promise<TransactionFormat> {
```

Comment:
did we want to add an optional `totalRecipientPercentAllocation` field? Defaults to 100, and we enforce that recipient allocations add up to `totalRecipientPercentAllocation`